common-rest/src/main/java/org/uniprot/api/rest/download/AbstractDownloadResultWriter.java

```diff
@@ -91,17 +98,32 @@ public void writeResult(
             context.setFields(request.getFields());
             context.setContentType(contentType);
 
-            if (supportedTypes.containsKey(contentType)) {
+            if (SUPPORTED_RDF_TYPES.containsKey(contentType)) {
                 Stream<String> rdfResponse =
-                        this.rdfStreamer.stream(ids, dataType, supportedTypes.get(contentType));
+                        this.rdfStreamer.stream(
+                                ids,
+                                dataType,
+                                SUPPORTED_RDF_TYPES.get(contentType),
+                                entries -> updateEntriesProcessed(downloadJob, entries));
                 context.setEntityIds(rdfResponse);
             } else if (contentType.equals(LIST_MEDIA_TYPE)) {
-                context.setEntityIds(ids);
+                context.setEntityIds(
+                        ids.map(
+                                id -> {
+                                    updateEntriesProcessed(downloadJob, 1);
+                                    return id;
+                                }));
             } else {
                 BatchStoreIterable<T> batchStoreIterable =
                         getBatchStoreIterable(ids.iterator(), storeRequest);
                 Stream<T> entities =
                         StreamSupport.stream(batchStoreIterable.spliterator(), false)
+                                .map(
+                                        entityCollection -> {
+                                            updateEntriesProcessed(
+                                                    downloadJob, entityCollection.size());
+                                            return entityCollection;
+                                        })
                                 .flatMap(Collection::stream)
                                 .onClose(
                                         () ->
```

Comment:
Put a try catch in the code where we are updating the processed count and continue with the remaining flow.